### ISSUE TITLE
Adding more statistics and more suggestions/feedback

### ISSUE THREAD

**Issue by fabsau** [Yooooomi/your_spotify#33]
Hi,
I am really enjoying this project and want to thank you for creating it.

I have a couple of ideas for further statistics:
- Display more than 7 best artist (via the modify menu)
- best artist Y-axis is songs listened right? I think play duration would be better
- top genres by song quantity/time duration (pie chart maybe?)
- the display of "best songs" is cool but one is just too little. maybe top10?

1. Then I have an administrative question - is it a big issue if I have the server (behind a reverse proxy, ssl enabled) publicly accessible from the internet?
2. I like that you added the disable registration, but maybe it would be better if only the first user can control the setting. I want to share this friends and I would like to be the only one able to control this setting.
3. On the artist graph I would suggest to have the same hover effect as on the "Listening repartition over day" graph for better clarity.
4. On all graphs when you hover the value of the x-axis does not have any unit, I think making it would be a visibility improvement.
5. On the desktop there is a not functional hamburger menu on top left. Does not bother me much, but I wanted to point it out.
6. The column "Played at" I would suggest displaying the timestamp in either am/pm or the european hh:mm. The hh(h)mm layout is a bit weird
7. Adding filter options in the history, like in a spreadsheet. Or at least a date picker to see what you heard on a specific date.
8. Dark theme :D
9. Some way to delete an user (when sharing this with other people will be needed)
10. Is it possible to poll in which country the artist is located? If so maybe a bar-chart or a world map to visualize the data

Again thank you for everything ♥

**Comment by Yooooomi** [comment]
Thanks a lot for your suggestions.

Displaying more than 7 artists is a good idea, I originally designed it so that it fits the screen but I could totally implement a "see fullscreen" feature to allow more space for a specific statistic and then display more than 7. 

I might also add the possibility to switch from tracks played to minutes played yes!
Top genres is very difficult to implement because Spotify is not accurate at all for the genres, you'll have for example: "belgium rapper", "belgium hip hop" and not basic genres such as "hip hop", "rap", or anything.

I will go through your questions one by one now:

1. The server uses cookies to authenticate to it there should be no problem with it being accessible publicly. The only threat is that it can be requested a lot, but like any other servers I guess.
2. I wanted to keep things simple, if you share it with friends they won't ruin everything right ? But yeah maybe later I will add more capabilities to this feature.
3. I clearly have work left on the hovers, I know they can sometimes be clearer.
4. Same answer as above, I still have a lot of work on the hovers.
5. Yes I will disable it on desktop
6. I originally did this so that we do not get confused with the length of the music, but I could definitely change that.
7. I am currently focusing on having relevant statistics and less on UX, but this is definitely something to implement in the future.
8. Yes, definitely, nothing more to add :)
9. Administrative tools are also a thing I need to work on. Originally I designed this app for myself so it misses easy buttons to manage the app definitely.
10. This! This idea is really good, I could definitely do this, I will definitely have a look at this idea!

Thanks for this very constructive issue, it gave me ideas and was very useful. Everything will be implemented sooner or later. I hope I answered all the points you were wondering about.

**Comment by Yooooomi** [comment]
In the last update that is currently building in the Github actions I took some of your feedback into account.
I think I'm going to close this issue since there is not one problem pointed out. I wrote the points that still need work in some place so that I can implement them in the future.

Thanks again for your feedback, was very helpful!

**Comment by fabsau** [comment]
Sorry for reopening the issue. I updated my instance and I like how you solved my suggestions! I have researched what similar tools (used to) offer and used this as inspiration for the following list:
- Most streamed album
- Picking a time frame for your data, you can't see data older than 1 year (right?)
- Resetting passwords of users (i guess this can be done inside the database)
- Most played playlists and podcasts (super low prio)
- this page makes a calculation how mainstream you are, it's kinda fun: https://mainstream.ghan.nl/
- this site uses a lot of different charts http://www.mymusichabits.com/visualize/fabsau . Maybe some of them might be helpful to you.

Feel free to instant close this issue and add my ideas to your list. I am grateful for you reading and appreciating my input. Lots of love 💕

**Comment by Yooooomi** [comment]
Hello again!
- Most streamed album should be easy to implement yes.
- You can already pick a time frame for the data using the `modify` button on the top right corner of each graph / certain cards
- Most played playlist would force me to save the context in which the song has been played and I'm not currently doing that. Given the fact that indeed it's not a priority plus the fact that I do not have the data, it is not near to be implemented imo.
- I already thought about fun statistics like this, but I do not think something relevant is possible since those services have access to huge amounts of data. For this example you gave, I would only have the popularity of a certain song then interpret this data to give you a mainstream coefficient.
- I tried your mymusichabits website but it seems to be down at the moment of writing, I'll check it out later.

I am currently thinking about creating a library allowing the user to edit the layout of his home page, changing the graphs, removing some and adding some other, because everyone has specific likings about certain statistics. So I think my next focus will be to add some personalization to the website since for the moment it's a place where statistics start to pile up to be honest.

Thanks again for your feedbacks.

**Comment by coffeverton** [comment]
@Yooooomi I think that when @fabsau mentioned the time frame, he meant that the app would download older data from Spotify.
For example, I set up the application yesterday, and so I can only see what I listened yesterday and today. It would be nice if we could see data from before we started using the app.

**Comment by Yooooomi** [comment]
Yeah I know. I opened an issue about that not a long ago. The thing is that the API doesn't give us result from much more days before. The solutions would be to download your account privacy data and search for each song through the Spotify API which can take a very long time.
You can check the issue at #26
